src/opentimelineio/clip.cpp

```diff
@@ -19,33 +27,74 @@ Clip::~Clip()
 MediaReference*
 Clip::media_reference() const noexcept
 {
-    return _media_reference;
+    auto active = _media_references.find(_active_media_reference);
+    return active == _media_references.end() || !active->second
+               ? nullptr
+               : active->second;
+}
+
+Clip::MediaReferences
+Clip::media_references() const noexcept
+{
+    MediaReferences result;
+    for (auto const& m: _media_references)
+    {
+        result.insert(
+            { m.first, dynamic_retainer_cast<MediaReference>(m.second) });
+    }
+    return result;
+}
+
+void
+Clip::set_media_references(MediaReferences const& media_references) noexcept
+{
+    _media_references.clear();
+    for (auto const& m: media_references)
+    {
+        _media_references[m.first] = m.second;
```

Comment:
Should we do this instead?
```
_media_references[m.first] = m.second ? m.second : new MissingReference
```

So that this:
```
clip.set_media_reference(nullptr);
```
behaves the same as this:
```
clip.set_media_references({"default", nullptr});
```

Reply:
What's the precedent elsewhere, within OTIO? 

As a C++ programmer, I intuitively expect set(nullptr) means whatever is set, is going to be cleared, in the general case.
If a new map is going to be created, I'd want that to consistently match other interfaces. @davidbaraff Have you got thoughts on this?

Reply:
If you set a nullptr (or None via the Python bindings) with `set_media_reference` on a Clip, it will result in the Clip's media reference being set to a MissingReference.  Personally, I find this a bit misleading and prefer explicit behaviour (ie if I wanted a MissingReference I would have done `clip.set_media_reference( new MissingReference() );`). 

Since set_media_reference is also called in the ctor, this effectively means a Clip can never have a null MediaReference, perhaps there is some historical reason for this?  Currently, all the code I can find is properly checking for nullptr even though it will never be the case.  For example in Clip::available_range:

```
    auto active_media = media_reference();
    if (!active_media)
    {
        if (error_status)
        {
            *error_status = ErrorStatus(
                ErrorStatus::CANNOT_COMPUTE_AVAILABLE_RANGE,
                "No media reference set on clip",
                this);
        }
        return TimeRange();
    }

    if (!active_media->available_range())
    {
        if (error_status)
        {
            *error_status = ErrorStatus(
                ErrorStatus::CANNOT_COMPUTE_AVAILABLE_RANGE,
                "No available_range set on media reference on clip",
                this);
        }
        return TimeRange();
    }
```

Reply:
I don't know the history, but it's consistent, thanks for looking that up.